sale_product_set/tests/test_product_set.py

```diff
@@ -166,3 +166,13 @@ def test_discount(self):
         )
         order_line.ensure_one()
         self.assertEqual(order_line.discount, set_line.discount)
+
+    def test_product_set_delete_in_wizard(self):
+        self.wizard = self._get_wiz()
+        self.wizard.product_set_line_ids -= self.wizard.product_set_line_ids[1]
```

Comment:
I bet this test pass even w/out your change because you set the value of the cache here, isn't it? :wink:

Reply:
Thanks. I fixed that now and agree with you. Could you look now, maybe you have more suggestions?